app/controllers/articles_controller.rb

```diff
@@ -1,15 +1,23 @@
 class ArticlesController < ApplicationController
-  before_action :set_article, only: %i[ show edit update destroy ]
+  before_action :set_search, :set_article, only: %i[ show edit update destroy ]
```

Comment:
no need to call `set_search` here, as it is already executed via before_action in ApplicationController